renderer_cocoa/lib/Cocoa_stubs.c

```diff
@@ -313,26 +365,41 @@ CAMLprim value ml_NSView_memoize(value id_v, value view_v)
   CAMLreturn(Val_unit);
 }
 
-CAMLprim value ml_NSView_free(value id_v)
+void ml_NSView_memoize(intnat id_v, View *view)
+{
+  [ml_Views setObject:view forKey:@(id_v)];
+}
+
+CAMLprim value ml_NSView_free_bc(value id_v)
 {
   CAMLparam1(id_v);
 
   [ml_Views removeObjectForKey:@(Int_val(id_v))];
   CAMLreturn(Val_unit);
 }
 
-CAMLprim value ml_NSView_addSubview(value view_v, value child_v)
+void ml_NSView_free(intnat id_v)
+{
+  [ml_Views removeObjectForKey:@(id_v)];
+}
+
+CAMLprim value ml_NSView_addSubview_bc(value view_v, value child_v)
 {
   CAMLparam2(view_v, child_v);
   View *view = View_val(view_v);
   View *child = View_val(child_v);
 
   [view addSubview:child];
 
-  CAMLreturn(Val_View(view));
+  CAMLreturn(Val_unit);
 }
 
-CAMLprim value ml_NSView_removeSubview(value child_v)
+void ml_NSView_addSubview(View *view, View *child)
+{
+  [view addSubview:child];
```

Comment:
The implementation can be the same for bytecode in such cases.

Reply:
At least  I believe so, although I don't know the true meaning of `CAMLParam` in byte code.